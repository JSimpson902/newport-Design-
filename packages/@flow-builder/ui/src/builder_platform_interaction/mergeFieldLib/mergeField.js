import { getResourceByUniqueIdentifier } from 'builder_platform_interaction/expressionUtils';
import { sanitizeGuid } from 'builder_platform_interaction/dataMutationLib';
import {
    getPropertiesForClass,
    getApexPropertyWithName
} from 'builder_platform_interaction/apexTypeLib';
import { describeExtension } from 'builder_platform_interaction/flowExtensionLib';
import {
    getExtensionParamDescriptionAsComplexTypeFieldDescription,
    getInvocableActionParamDescriptionAsComplexTypeFieldDescription
} from 'builder_platform_interaction/complexTypeLib';
import {
    isComplexType,
    FLOW_DATA_TYPE
} from 'builder_platform_interaction/dataTypeLib';
import {
    fetchFieldsForEntity,
    getEntityFieldWithApiName
} from 'builder_platform_interaction/sobjectLib';
import { isLookupTraversalSupported as isLookupTraversalSupportedByProcessType } from 'builder_platform_interaction/processTypeLib';
import { isLookupTraversalSupported as isLookupTraversalSupportedByTriggerType } from 'builder_platform_interaction/triggerTypeLib';
import { fetchDetailsForInvocableAction } from 'builder_platform_interaction/invocableActionLib';

/**
 * Whether or not lookup traversal is supported in this flow
 * @param {String} processType the current process type
 * @param {String} triggerType the current trigger type
 * @returns {Boolean} true if lookup traversal is supported, false otherwise
 */
export const isLookupTraversalSupported = (processType, triggerType) => {
    return (
        isLookupTraversalSupportedByProcessType(processType) &&
        isLookupTraversalSupportedByTriggerType(triggerType)
    );
};

/**
 * Resolve the reference given its identifier.
 * The identifier can be a resource/element guid ('a4451815-988d-4f17-883d-64b6ad9fab7e'), a global constant identifier ('$GlobalConstant.EmptyString'), a system variable identifier or
 * a field reference identifier ('a4451815-988d-4f17-883d-64b6ad9fab7e.Account.User.Name')
 *
 * @returns {Promise<Object[]|undefined>} Array with first item being the element (resource/global constant ...) and next items being the fields. Returns undefined if not a valid reference
 */
export function resolveReferenceFromIdentifier(identifier) {
    const elementOrResource = getResourceByUniqueIdentifier(identifier);
    if (!elementOrResource) {
        return Promise.resolve();
    }
    if (isComplexType(elementOrResource.dataType)) {
        const { fieldNames } = sanitizeGuid(identifier);
        return resolveComplexTypeReference(elementOrResource, fieldNames);
    }
    return Promise.resolve([elementOrResource]);
}

function resolveComplexTypeReference(flowResource, fieldNames) {
    if (!fieldNames || fieldNames.length === 0) {
        return Promise.resolve([flowResource]);
    }
    let fieldsPromise;
    if (flowResource.dataType === FLOW_DATA_TYPE.SOBJECT.value) {
        fieldsPromise = resolveEntityFieldReference(
            flowResource.subtype,
            fieldNames
        );
    } else if (flowResource.dataType === FLOW_DATA_TYPE.APEX.value) {
        fieldsPromise = resolveApexPropertyReference(
            flowResource.subtype,
            fieldNames
        );
    } else if (
        flowResource.dataType ===
        FLOW_DATA_TYPE.LIGHTNING_COMPONENT_OUTPUT.value
    ) {
        fieldsPromise = resolveLightningComponentOutputReference(
            flowResource.extensionName,
            fieldNames
        );
    } else if (flowResource.dataType === FLOW_DATA_TYPE.ACTION_OUTPUT.value) {
        fieldsPromise = resolveActionOutputReference(flowResource, fieldNames);
    }
    if (fieldsPromise) {
        return fieldsPromise.then(fields => {
            return fields ? [flowResource, ...fields] : undefined;
        });
    }
    return Promise.resolve(undefined);
}

function resolveApexPropertyReference(clazz, fieldNames) {
    if (fieldNames.length === 0) {
        return Promise.resolve([]);
    }
    const [fieldName, ...remainingFieldNames] = fieldNames;
    return Promise.resolve(getPropertiesForClass(clazz)).then(properties => {
        const property = getApexPropertyWithName(properties, fieldName);
        if (!property) {
            return undefined;
        }
        if (remainingFieldNames.length > 0) {
            if (property.dataType === FLOW_DATA_TYPE.APEX.value) {
                return resolveApexPropertyReference(
                    property.subtype,
                    remainingFieldNames
                ).then(fields => {
                    return fields ? [property, ...fields] : undefined;
                });
            } else if (property.dataType === FLOW_DATA_TYPE.SOBJECT.value) {
                return resolveEntityFieldReference(
                    property.subtype,
                    remainingFieldNames
                ).then(fields => {
                    return fields ? [property, ...fields] : undefined;
                });
            }
            return undefined;
        }
        return [property];
    });
}

function resolveEntityFieldReference(entityName, fieldNames) {
    if (fieldNames.length === 0) {
        return Promise.resolve([]);
    }
    const [fieldName, ...remainingFieldNames] = fieldNames;
    return fetchFieldsForEntity(entityName, { disableErrorModal: true }).then(
        fields => {
            if (remainingFieldNames.length > 0) {
                const {
                    relationshipName,
                    specificEntityName
                } = getPolymorphicRelationShipName(fieldName);
                const field = getEntityFieldWithRelationshipName(
                    fields,
                    relationshipName
                );
                if (!field) {
                    return undefined;
                }
                const referenceToName = getReferenceToName(
                    field,
                    specificEntityName
                );
                return resolveEntityFieldReference(
                    referenceToName,
                    remainingFieldNames
                ).then(remainingFields => {
                    return remainingFields
                        ? [field, ...remainingFields]
                        : undefined;
                });
            }
            const field = getEntityFieldWithApiName(fields, fieldName);
            if (!field) {
                return undefined;
            }
            return [field];
        }
    );
}

function resolveLightningComponentOutputReference(extensionName, fieldNames) {
    if (fieldNames.length === 0) {
        return Promise.resolve([]);
    }
    const [fieldName, ...remainingFieldNames] = fieldNames;
    return fetchExtensionOutputParameters(extensionName).then(
        outputParameters => {
            const field = outputParameters[fieldName];
            if (!field) {
                return undefined;
            }
            if (remainingFieldNames.length > 0) {
                return resolveApexPropertyOrEntityFieldReference(
                    field,
                    remainingFieldNames
                ).then(remainingFields => {
                    return remainingFields
                        ? [field, ...remainingFields]
                        : undefined;
                });
            }
            return [field];
        }
    );
}

function resolveApexPropertyOrEntityFieldReference(field, fieldNames) {
    if (field.dataType === FLOW_DATA_TYPE.APEX.value) {
        return resolveApexPropertyReference(field.subtype, fieldNames);
    } else if (field.dataType === FLOW_DATA_TYPE.SOBJECT.value) {
        return resolveEntityFieldReference(field.subtype, fieldNames);
    }
    return Promise.resolve(undefined);
}

function resolveActionOutputReference({ actionType, actionName }, fieldNames) {
    if (fieldNames.length === 0) {
        return Promise.resolve([]);
    }
    const [fieldName, ...remainingFieldNames] = fieldNames;
    return fetchActionOutputParameters({ actionType, actionName }).then(
        outputParameters => {
            const field = outputParameters[fieldName];
            if (!field) {
                return undefined;
            }
            if (remainingFieldNames.length > 0) {
                return resolveApexPropertyOrEntityFieldReference(
                    field,
                    remainingFieldNames
                ).then(remainingFields => {
                    return remainingFields
                        ? [field, ...remainingFields]
                        : undefined;
                });
            }
            return [field];
        }
    );
}

export function getReferenceToName(
    field,
    specificEntityName
) {
    let referenceToName;
    if (specificEntityName) {
        if (
            !field.isPolymorphic ||
            !entityFieldIncludesReferenceToName(field, specificEntityName)
        ) {
            return undefined;
        }
        referenceToName = specificEntityName;
    } else {
        if (field.isPolymorphic) {
            return undefined;
        }
        referenceToName = field.referenceToNames[0];
    }
    return referenceToName;
}

function entityFieldIncludesReferenceToName(entityField, referenceToName) {
    referenceToName = referenceToName.toLowerCase();
    return entityField.referenceToNames.some(
        fieldReferenceToName =>
            fieldReferenceToName.toLowerCase() === referenceToName
    );
}

export function getPolymorphicRelationShipName(fieldName) {
    const index = fieldName.indexOf(':');
    if (index < 0) {
        return { relationshipName: fieldName };
    }
    return {
        relationshipName: fieldName.substr(0, index),
        specificEntityName: fieldName.substr(index + 1)
    };
}

export function getEntityFieldWithRelationshipName(fields, relationshipName) {
    relationshipName = relationshipName.toLowerCase();
    for (const apiName in fields) {
        if (fields.hasOwnProperty(apiName)) {
            const field = fields[apiName];
            if (field.isSpanningAllowed) {
                if (
                    field.relationshipName &&
                    relationshipName === field.relationshipName.toLowerCase()
                ) {
                    return fields[apiName];
                } else if (
                    relationshipName === field.apiName.toLowerCase() &&
                    field.isCustom
                ) {
                    return fields[apiName];
                }
            }
        }
    }
    return undefined;
}

function getExtensionOutputParamDescriptions(extension) {
    return extension.outputParameters.reduce((acc, parameter) => {
        acc[
            parameter.apiName
        ] = getExtensionParamDescriptionAsComplexTypeFieldDescription(
            parameter
        );
        return acc;
    }, {});
}

function fetchExtensionOutputParameters(extensionName) {
    return describeExtension(extensionName, { disableErrorModal: true }).then(
        extension => getExtensionOutputParamDescriptions(extension)
    );
}

function getActionOutputParameterDescriptions(details) {
    return details.parameters
        .filter(parameter => parameter.isOutput === true)
        .reduce((acc, parameter) => {
            acc[
                parameter.name
            ] = getInvocableActionParamDescriptionAsComplexTypeFieldDescription(
                parameter
            );
            return acc;
        }, {});
}

function fetchActionOutputParameters({ actionType, actionName }) {
    return fetchDetailsForInvocableAction(
        { actionType, actionName },
        {
            disableErrorModal: true
        }
    ).then(details => getActionOutputParameterDescriptions(details));
}
