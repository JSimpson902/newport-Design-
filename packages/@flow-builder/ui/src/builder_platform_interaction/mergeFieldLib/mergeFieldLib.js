export {
    validateMergeField,
    validateTextWithMergeFields,
    isTextWithMergeFields
} from './mergeFieldValidation';
export {
    resolveReferenceFromIdentifier,
    isLookupTraversalSupported
} from './mergeField';
export { loadReferencesIn } from './references';
