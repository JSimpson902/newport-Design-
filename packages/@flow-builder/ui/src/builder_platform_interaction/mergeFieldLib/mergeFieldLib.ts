export { isLookupTraversalSupported, resolveReferenceFromIdentifier } from './mergeField';
export { isTextWithMergeFields, validateMergeField, validateTextWithMergeFields } from './mergeFieldValidation';
export { loadReferencesIn } from './references';
