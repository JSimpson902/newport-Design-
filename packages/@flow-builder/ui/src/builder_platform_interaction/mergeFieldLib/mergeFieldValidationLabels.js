import notAValidMergeField from '@salesforce/label/FlowBuilderMergeFieldValidation.notAValidMergeField';
import unknownResource from '@salesforce/label/FlowBuilderMergeFieldValidation.unknownResource';
import resourceCannotBeUsedAsMergeField from '@salesforce/label/FlowBuilderMergeFieldValidation.resourceCannotBeUsedAsMergeField';
import unknownRecordField from '@salesforce/label/FlowBuilderMergeFieldValidation.unknownRecordField';
import globalConstantsNotAllowed from '@salesforce/label/FlowBuilderMergeFieldValidation.globalConstantsNotAllowed';
import invalidDataType from '@salesforce/label/FlowBuilderMergeFieldValidation.invalidDataType';
import genericErrorMessage from '@salesforce/label/FlowBuilderCombobox.genericErrorMessage';
import invalidPolymorphicRecordField from '@salesforce/label/FlowBuilderMergeFieldValidation.invalidPolymorphicRecordField';

export const LABELS = {
    genericErrorMessage,
    notAValidMergeField,
    unknownResource,
    resourceCannotBeUsedAsMergeField,
    unknownRecordField,
    globalConstantsNotAllowed,
    invalidDataType,
    invalidPolymorphicRecordField
};
