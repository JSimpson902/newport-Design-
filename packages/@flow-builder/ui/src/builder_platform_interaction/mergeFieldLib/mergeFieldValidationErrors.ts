// @ts-nocheck
import { LABELS } from './mergeFieldValidationLabels';
import { format } from 'builder_platform_interaction/commonUtils';

const VALIDATION_ERROR_TYPE = {
    INVALID_MERGE_FIELD: 'notAValidMergeField',
    INVALID_GLOBAL_CONSTANT: 'invalidGlobalConstant',
    INVALID_GLOBAL_VARIABLE: 'invalidGlobalVariable',
    UNKNOWN_MERGE_FIELD: 'unknownMergeField',
    WRONG_DATA_TYPE: 'wrongDataType',
    MAXIMUM_LEVEL_REACHED: 'maximumLevelReached'
};

function validationError(errorType, message, startIndex, endIndex) {
    return {
        errorType,
        message,
        startIndex,
        endIndex
    };
}

export function mergeFieldNotAllowed(startIndex, endIndex) {
    return validationError(VALIDATION_ERROR_TYPE.INVALID_MERGE_FIELD, LABELS.genericErrorMessage, startIndex, endIndex);
}

export function maximumNumberOfLevelsReached(startIndex, endIndex) {
    return validationError(
        VALIDATION_ERROR_TYPE.MAXIMUM_LEVEL_REACHED,
        LABELS.maximumNumberOfLevelsReached,
        startIndex,
        endIndex
    );
}

export function globalConstantsNotAllowed(startIndex, endIndex) {
    return validationError(
        VALIDATION_ERROR_TYPE.INVALID_MERGE_FIELD,
        LABELS.globalConstantsNotAllowed,
        startIndex,
        endIndex
    );
}

export function invalidGlobalConstant(mergeField, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.genericErrorMessage, mergeField);
    return validationError(VALIDATION_ERROR_TYPE.INVALID_GLOBAL_CONSTANT, validationErrorLabel, startIndex, endIndex);
}

export function invalidMergeField(mergeField, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.notAValidMergeField, mergeField);
    return validationError(VALIDATION_ERROR_TYPE.INVALID_MERGE_FIELD, validationErrorLabel, startIndex, endIndex);
}

export function invalidDataType(startIndex, endIndex) {
    return validationError(VALIDATION_ERROR_TYPE.WRONG_DATA_TYPE, LABELS.invalidDataType, startIndex, endIndex);
}

export function invalidGlobalVariable(mergeField, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.genericErrorMessage, mergeField);
    return validationError(VALIDATION_ERROR_TYPE.INVALID_GLOBAL_VARIABLE, validationErrorLabel, startIndex, endIndex);
}

export function unknownResource(resourceName, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.unknownResource, resourceName);
    return validationError(VALIDATION_ERROR_TYPE.UNKNOWN_MERGE_FIELD, validationErrorLabel, startIndex, endIndex);
}

export function resourceCannotBeUsedAsMergeField(resourceName, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.resourceCannotBeUsedAsMergeField, resourceName);
    return validationError(VALIDATION_ERROR_TYPE.WRONG_DATA_TYPE, validationErrorLabel, startIndex, endIndex);
}

export function unknownRecordField(entityName, fieldName, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.unknownRecordField, fieldName, entityName);
    return validationError(VALIDATION_ERROR_TYPE.UNKNOWN_MERGE_FIELD, validationErrorLabel, startIndex, endIndex);
}

export function invalidPolymorphicRecordFieldReference(entityName, polymorphicReference, startIndex, endIndex) {
    const validationErrorLabel = format(LABELS.invalidPolymorphicRecordField, entityName, polymorphicReference);
    return validationError(VALIDATION_ERROR_TYPE.UNKNOWN_MERGE_FIELD, validationErrorLabel, startIndex, endIndex);
}
