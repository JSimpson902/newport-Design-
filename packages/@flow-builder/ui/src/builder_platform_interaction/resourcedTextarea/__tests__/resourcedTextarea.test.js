import {createElement} from 'lwc';
import ResourcedTextarea from '../resourcedTextarea';
import { getShadowRoot } from 'lwc-test-utils';
import { validateTextWithMergeFields } from 'builder_platform_interaction/mergeFieldLib';

const label = 'Help Text';
const changeEventName = 'change';
const itemText = '{!var}';
const resourcePickerVal = 'initialValue';

const createComponentUnderTest = (props) => {
    const el = createElement('builder_platform_interaction-resourced-textarea', {
        is: ResourcedTextarea
    });
    Object.assign(el, props);
    document.body.appendChild(el);
    return el;
};

jest.mock('builder_platform_interaction/mergeFieldLib', () => {
    return {
        validateTextWithMergeFields: jest.fn().mockResolvedValue([]),
    };
});

const selectors = {
    label: 'label',
    abbr: 'abbr',
    textarea: 'textarea',
    ferovResourcePicker: 'builder_platform_interaction-ferov-resource-picker'
};

describe('Resourced text area label', () => {
    it('Should have the asterisk when required', () => {
        const element = createComponentUnderTest({required: true, label});
        const fullLabel = getShadowRoot(element).querySelector(selectors.label);
        expect(fullLabel.textContent).toMatch(label);
        const asterisk = fullLabel.querySelector(selectors.abbr);
        expect(asterisk.textContent).toMatch('*');
    });
    it('Should not have the asterisk when not required', () => {
        const element = createComponentUnderTest({required: false, label});
        const fullLabel = getShadowRoot(element).querySelector(selectors.label);
        expect(fullLabel.textContent).toMatch(label);
        const asterisk = fullLabel.querySelector(selectors.abbr);
        expect(asterisk).toBeFalsy();
    });
});

function verifyItemInsertion(existingText, selectionStart, selectionEnd, expectedFinalText, expectedFinalCursorPosition) {
    const element = createComponentUnderTest({value: existingText});
    const textarea = getShadowRoot(element).querySelector(selectors.textarea);
    textarea.setSelectionRange(selectionStart, selectionEnd);
    const ferovResourcePicker = getShadowRoot(element).querySelector(selectors.ferovResourcePicker);
    const changeEventCallback = jest.fn();
    element.addEventListener(changeEventName, changeEventCallback);
    ferovResourcePicker.value = resourcePickerVal;
    const itemSelectedEvent = new CustomEvent('itemselected', {detail: {item: {displayText: itemText}}});
    ferovResourcePicker.dispatchEvent(itemSelectedEvent);
    return Promise.resolve().then(() => {
        expect(element.value).toBe(existingText);
        expect(textarea.value).toBe(expectedFinalText);
        expect(textarea.selectionStart).toBe(expectedFinalCursorPosition);
        expect(textarea.selectionEnd).toBe(expectedFinalCursorPosition);
        expect(ferovResourcePicker.value).toBeNull();
        expect(changeEventCallback).toHaveBeenCalled();
    });
}

describe('Item selection from the resource picker', () => {
    it('Should insert the item when there is no text', () => {
        return verifyItemInsertion('', 0, 0, itemText, itemText.length);
    });
    it('Should insert the item when the cursor is at the beginning of text', () => {
        const existingText = 'Sample Text';
        return verifyItemInsertion(existingText, 0, 0, itemText + existingText, itemText.length);
    });
    it('Should insert the item when the cursor is in the middle of text', () => {
        return verifyItemInsertion('Sample Text', 6, 6, 'Sample{!var} Text', 12);
    });
    it('Should insert the item when the cursor is at the end of text', () => {
        const existingText = 'Sample Text';
        const finalText = existingText + itemText;
        return verifyItemInsertion(existingText, existingText.length,
            existingText.length, finalText, finalText.length);
    });
    it('Should insert the item when the cursor is on a new line', () => {
        const existingText = 'Sample Text\n';
        const finalText = existingText + itemText;
        return verifyItemInsertion(existingText, existingText.length,
            existingText.length, finalText, finalText.length);
    });
    it('Should replace selected text with the inserted item', () => {
        const existingText = 'Sample Text';
        const finalText = 'Sample {!var}';
        return verifyItemInsertion(existingText, 7, existingText.length, finalText, finalText.length);
    });
    it('Should not do anything if it has a next item', () => {
        const existingText = 'Sample Text';
        const element = createComponentUnderTest({value: existingText});
        const textarea = getShadowRoot(element).querySelector(selectors.textarea);
        textarea.setSelectionRange(0, 0);
        const itemSelectedEvent2 = new CustomEvent('itemselected', {detail: {item: {hasNext: true}}});
        const ferovResourcePicker = getShadowRoot(element).querySelector(selectors.ferovResourcePicker);
        const changeEventCallback = jest.fn();
        element.addEventListener(changeEventName, changeEventCallback);
        ferovResourcePicker.value = resourcePickerVal;
        ferovResourcePicker.dispatchEvent(itemSelectedEvent2);
        return Promise.resolve().then(() => {
            expect(element.value).toBe(existingText);
            expect(textarea.value).toBe(existingText);
            expect(textarea.selectionStart).toBe(0);
            expect(textarea.selectionEnd).toBe(0);
            expect(ferovResourcePicker.value).toBe(resourcePickerVal);
            expect(changeEventCallback).not.toHaveBeenCalled();
        });
    });
});

describe('Events from the textarea', () => {
    let eventCallback;

    const expectValueChangedEventWithValue = (value, error) => {
        expect(eventCallback).toHaveBeenCalled();
        expect(eventCallback.mock.calls[0][0]).toMatchObject({ detail: { value, error } });
    };

    beforeEach(async () => {
        eventCallback = jest.fn();
    });

    it('Should fire change event on blur', async () => {
        const resourcedTextarea = createComponentUnderTest({value: '1+1'});
        resourcedTextarea.addEventListener('change', eventCallback);

        const textarea = getShadowRoot(resourcedTextarea).querySelector(selectors.textarea);
        textarea.value = '2+2';
        textarea.dispatchEvent(new CustomEvent('blur'));
        await Promise.resolve();
        expectValueChangedEventWithValue('2+2', null);
    });

    it('Should fire change event on blur with validation errors', async () => {
        const resourcedTextarea = createComponentUnderTest({value: '{!unknownMergeField}'});
        resourcedTextarea.addEventListener('change', eventCallback);

        const validationError = {
            errorType : 'errorType',
            message : 'errorMessage',
            startIndex : 0,
            endIndex : 0
        };
        validateTextWithMergeFields.mockResolvedValue([validationError]);

        const textarea = getShadowRoot(resourcedTextarea).querySelector(selectors.textarea);
        textarea.dispatchEvent(new CustomEvent('blur'));
        await Promise.resolve();
        expectValueChangedEventWithValue('{!unknownMergeField}', validationError.message);
    });
});