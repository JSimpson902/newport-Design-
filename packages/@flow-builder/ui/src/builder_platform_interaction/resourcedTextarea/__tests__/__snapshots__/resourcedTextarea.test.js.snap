// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resourced text area "rich text/ plain text mode switch" NOT displayed by default (snapshot) 1`] = `
<builder_platform_interaction-resourced-textarea>
  #shadow-root(open)
    <div
      class="property-input slds-form-element"
    >
      <div
        class="container slds-grid slds-grid_vertical"
      >
        <div
          class="divParentResourcePickerModeOptions slds-p-around_x-small slds-p-bottom_xx-small slds-button-group-list"
        >
          <div
            class="divResourcePickerFulllWidth"
          >
            <builder_platform_interaction-ferov-resource-picker>
              #shadow-root(open)
            </builder_platform_interaction-ferov-resource-picker>
          </div>
        </div>
        <textarea
          class="slds-textarea slds-input textarea-input"
        />
      </div>
    </div>
</builder_platform_interaction-resourced-textarea>
`;

exports[`Resourced text area "rich text/ plain text mode switch" NOT displayed if "plainTextAvailable" API set to false 1`] = `
<builder_platform_interaction-resourced-textarea>
  #shadow-root(open)
    <div
      class="property-input slds-form-element"
    >
      <div
        class="container slds-grid slds-grid_vertical"
      >
        <div
          class="divParentResourcePickerModeOptions slds-p-around_x-small slds-p-bottom_xx-small slds-button-group-list"
        >
          <div
            class="divResourcePickerFulllWidth"
          >
            <builder_platform_interaction-ferov-resource-picker>
              #shadow-root(open)
            </builder_platform_interaction-ferov-resource-picker>
          </div>
        </div>
        <textarea
          class="slds-textarea slds-input textarea-input"
        />
      </div>
    </div>
</builder_platform_interaction-resourced-textarea>
`;

exports[`Resourced text area "rich text/ plain text mode switch" displayed if "plainTextAvailable" API set to true 1`] = `
<builder_platform_interaction-resourced-textarea>
  #shadow-root(open)
    <div
      class="property-input slds-form-element"
    >
      <div
        class="container slds-grid slds-grid_vertical"
      >
        <div
          class="divParentResourcePickerModeOptions slds-p-around_x-small slds-p-bottom_xx-small slds-button-group-list"
        >
          <div
            class="divResourcePickerPartialWidth"
          >
            <builder_platform_interaction-ferov-resource-picker>
              #shadow-root(open)
            </builder_platform_interaction-ferov-resource-picker>
          </div>
          <builder_platform_interaction-rich-text-plain-text-switch>
            #shadow-root(open)
              <div
                class="divButtonMenuPlainRichText"
              >
                <lightning-button-menu>
                  #shadow-root(open)
                    <slot />
                  <lightning-menu-item>
                    #shadow-root(open)
                      <span
                        class="slds-truncate"
                      >
                        FlowBuilderResourcedRichTextEditor.richText
                      </span>
                  </lightning-menu-item>
                  <lightning-menu-item>
                    #shadow-root(open)
                      <span
                        class="slds-truncate"
                      >
                        <lightning-primitive-icon>
                          #shadow-root(open)
                            <svg
                              data-key="utility:check"
                            >
                              <use
                                xlink:href="null"
                              />
                            </svg>
                        </lightning-primitive-icon>
                        FlowBuilderResourcedRichTextEditor.plainText
                      </span>
                  </lightning-menu-item>
                </lightning-button-menu>
              </div>
          </builder_platform_interaction-rich-text-plain-text-switch>
        </div>
        <textarea
          class="slds-textarea slds-input textarea-input"
        />
      </div>
    </div>
</builder_platform_interaction-resourced-textarea>
`;
